src/orb/orb.h

```diff
@@ -0,0 +1,158 @@
+#ifndef ORB_H
+#define ORB_H
+
+#include "bit_pattern_31.h"
+#include <imgproc/imgproc.h>
+#include <iostream>
+#include <jslog/jslog.h>
+#include <keypoint_t/keypoint_t.h>
+#include <matrix_t/matrix_t.h>
+//#include <emscripten/val.h>
+#include <types/types.h>
+
+namespace jsfeat {
+
+class orb {
+public:
+  orb() {
+    H = new matrix_t(3, 3, Types::F32_t | Types::C1_t, Float32Array);
+    H->allocate();
+    patch_img = new matrix_t(32, 32, Types::U8_t | Types::C1_t, Uint8Array);
+    patch_img->allocate();
+  }
+
+  void describe(uintptr_t inputSrc, uintptr_t inputCorners, int count,
+                uintptr_t inputDescriptors) {
+    auto src = reinterpret_cast<matrix_t *>(inputSrc);
+    auto corners = reinterpret_cast<keypoint_t *>(inputCorners);
```

Comment:
This can not work because it's expected to pass an array of **keypoint_t**. Better maybe make inputCorners as an emscripten::val or a struct and export it with embind as a `value_object`.

Reply:
with emscripten::vecFromJSArray i can load an array of **keypoint_t**. I have no errors testing the orb_test example. Though it is required more testing.